src/main/java/toughasnails/temperature/BlockTemperatureData.java

```diff
@@ -0,0 +1,14 @@
+package toughasnails.temperature;
+
+public class BlockTemperatureData {
+	public String blockName;
+	public float blockBaseTemperature;
+	public BlockTemperatureStateData[] stateTemperatures;
```

Comment:
It seems you've recreated the properties of block states rather than just working natively with actual IBlockStates. Here's what we do in BoP for example: https://github.com/Glitchfiend/BiomesOPlenty/blob/BOP-1.11.x-6.0.x/src/main/java/biomesoplenty/common/util/config/BOPConfig.java#L447-L512